pandas/core/categorical.py

```diff
@@ -1995,14 +1995,18 @@ def is_dtype_equal(self, other):
 
         Parameters
         ----------
-        other : Categorical
+        other : Categorical, Series
 
         Returns
         -------
         are_equal : boolean
         """
-
         try:
+            from pandas.core.series import Series
+
+            if isinstance(other, Series):
+                other = Categorical(other)
```

Comment:
Thanks for your guidance.

It seems that the following code
```python
if not isinstance(other, Categorical) and not isinstance(CategoricalIndex):
    other = Categorical(other)
```
will lead to the failure of following test case
```python
assert not c1.is_dtype_equal(Index(list('aabca')))
```